utils/misc/misc.go

```diff
@@ -923,6 +923,16 @@ func FastUUID() uuid.UUID {
 	return uuid.FromBytesOrNil(b)
 }
 
+func HasAWSRoleARNInConfig(configMap map[string]interface{}) bool {
+	if configMap["iamRoleARN"] == nil {
+		return false
+	}
+	if configMap["iamRoleARN"].(string) == "" {
```

Comment:
To avoid a potential panic in case this isn't a string, maybe it makes more sense to do:
```suggestion
	iamRoleARN, ok := configMap["iamRoleARN"].(string)
	if !ok {
		return false
	}
	if iamRoleARN == "" {
```